File: app/src/main/java/org/citra/citra_emu/model/GameInfo.java
package org.citra.citra_emu.model;

/**
 * 3DS 游戏元信息 —— 与上游 org.citra.citra_emu.model.GameInfo（Kotlin 类）
 * 保持一致的 JVM 形状：构造器 GameInfo(String path) → native initialize(path)
 * 返回核心侧元信息指针，其余 getter/setter 全部经该指针读写。
 *
 * NesStation 用它做：
 *  - 3DS 游戏扫描元数据（标题/公司/图标/区域）
 *  - ★ 加密检测（isEncrypted）—— 引导用户导入 aes_keys.txt 或先解密
 */
public final class GameInfo {

    private final long pointer;

    public GameInfo(String path) {
        this.pointer = initialize(path);
    }

    private static final native long initialize(String path);

    protected final native void finalize();

    public final native boolean isValid();

    public final native String getTitle();

    public final native String getCompany();

    public final native long getTitleID();

    public final native int[] getIcon();

    public final native String getRegions();

    public final native String getFileType();

    public final native boolean isEncrypted();

    public final native boolean isSystemTitle();

    public final native boolean getIsVisibleSystemTitle();

    public final native boolean getIsInsertable();
}

File: app/src/main/java/org/citra/citra_emu/model/GameInfo.kt
package org.citra.citra_emu.model

import androidx.annotation.Keep

/**
 * Azahar GameInfo —— JNI 契约桩。
 *
 * 原生侧（src/android/app/src/main/jni/id_cache.cpp）读取 `pointer:J` 字段
 * （原版游戏库列表把 native GameInfo* 包在该字段里）。NesStation 游戏库走
 * 自有 RomStore，不实例化本类，仅保证 FindClass / GetFieldID 成功。
 */
@Keep
class GameInfo {
    @JvmField
    var pointer: Long = 0L
}
